examples/kafka_spark_demo/feature_repo/streaming_features_tutorial.ipynb

```diff
@@ -0,0 +1,877 @@
+      "\t|                  |            modules            ||   artifacts   |\n",
+      "\t|       conf       | number| search|dwnlded|evicted|| number|dwnlded|\n",
+      "\t---------------------------------------------------------------------\n",
+      "\t|      default     |   9   |   0   |   0   |   0   ||   9   |   0   |\n",
+      "\t---------------------------------------------------------------------\n",
+      ":: retrieving :: org.apache.spark#spark-submit-parent-be0e71e9-46f4-47a3-a6fc-3661ae6b71c8\n",
+      "\tconfs: [default]\n",
+      "\t0 artifacts copied, 9 already retrieved (0kB/9ms)\n",
+      "22/06/17 11:32:59 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable\n",
+      "Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties\n",
+      "Setting default log level to \"WARN\".\n",
+      "To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "os.environ['PYSPARK_SUBMIT_ARGS'] = \"--packages=org.apache.spark:spark-sql-kafka-0-10_2.12:3.0.0 pyspark-shell\"\n",
+    "spark = SparkSession.builder.master(\"local\").appName(\"feast-spark\").getOrCreate()\n",
+    "spark.conf.set(\"spark.sql.shuffle.partitions\", 5)\n",
+    "\n",
+    "from typing import List"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The sample code requires three things to launch a job: a config, a stream feature view, and a function to specify where the job output will be written. Here we import the sample code and set the configuration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from feast.infra.contrib.stream_processor import ProcessorConfig\n",
+    "from feast.infra.contrib.spark_kafka_processor import SparkProcessorConfig\n",
+    "from feast.infra.contrib.stream_processor import get_stream_processor_object\n",
+    "\n",
+    "spark_config = SparkProcessorConfig(mode=\"spark\", source=\"kafka\", spark_session=spark, processing_time=\"30 seconds\", query_timeout=15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We define the method that will write data from our Kafka stream to the online store. We include several `print` statements so we can monitor what's going on."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def write_stream_row(\n",
+    "    fs: FeatureStore, \n",
+    "    feature_view: str,\n",
+    "    row: pd.DataFrame,\n",
+    "    join_keys: List[str],\n",
+    "    input_timestamp_field: str,\n",
+    "    output_timestamp_column: str = \"\",\n",
+    "):\n",
+    "    row = (\n",
```

Comment:
Uhh is this pulling out a single row for the write? Won't the dataframe contain multiple rows?

Reply:
Oh this should be `rows`

Reply:
Can you explain why we're doign all of this stuff here to create `rows`? Why are we grouping by or sorting values etc etc?

Reply:
good catch again